code/core/src/core-server/utils/StoryIndexGenerator.ts

```diff
@@ -792,10 +803,18 @@ export class StoryIndexGenerator {
     });
     this.lastIndex = null;
     this.lastError = null;
+    this.invalidationListeners.forEach((listener) => listener());
   }
 
-  invalidate(specifier: NormalizedStoriesSpecifier, importPath: Path, removed: boolean) {
+  invalidate(importPath: Path, removed: boolean) {
```

Comment:
The reason I'm removing the `specifier` from the `invalidate` signature, is because it meant that the StoryIndexGenerator implicitly relied on the specifiers to be referential equal, as they were used as keys for Map lookups. This was not obvious, so during my refactor I was creating multiple specifiers with `normalizeStories`, and they would not be referential equal, which I spent a lot of time debugging.

This change means we rely on the `importPath` to lookup the specifier, similar to what used to happen in `watchStorySpecifier`. `importPath` is just a string, so equality is simpler.